tests/test_ampOffset.py

```diff
@@ -351,7 +351,7 @@ def runAmpOffsetWithBackground(self, valueType, rampBackground=False):
             self.assertAlmostEqual(
                 np.std(pedestals - approximatePedestals),
                 measuredSigma[weightType][valueType],
-                12,
+                4,  # DO NOT MERGE
```

Comment:
@enourbakhsh to confirm, you are okay with this being only 4 decimal places?

Reply:
Do we know if it’s just one number causing it or if all of them are affected? I’m just curious, could there be any type conversion happening somewhere that’s introducing precision loss?

Reply:
```tests/test_ampOffset.py::AmpOffsetTest::testAmpOffsetWithRampBackground - AssertionError: np.float64(6.08989306958048) != 6.089900455983161 within 8 places (np.float64(7.386402680609194e-06) difference)```

Reply:
so it's weighted artificial that is changing.

Reply:
I'd make the number of decimal places conditional on the `rampBackground` parameter because the approximations are less reliable when using ramp backgrounds. Does this fix it?
```python
atol = 4 if rampBackground else 12
```

Reply:
```diff
diff --git a/tests/test_ampOffset.py b/tests/test_ampOffset.py
index 4303b97..c405ae3 100644
--- a/tests/test_ampOffset.py
+++ b/tests/test_ampOffset.py
@@ -351,7 +351,7 @@ class AmpOffsetTest(lsst.utils.tests.TestCase):
             self.assertAlmostEqual(
                 np.std(pedestals - approximatePedestals),
                 measuredSigma[weightType][valueType],
-                4,  # DO NOT MERGE
+                4 if rampBackground else 12,
             )
             if valueType == "artificial":
                 if not applyWeights:
```
does work.

Reply:
Great! I'd be happy with this change.

Reply:
Changed. I can merge once the full PR is reviewed.